services/publish.webmaker.org/lib/remix.js

```diff
@@ -22,10 +24,27 @@ function injectMetadata(html, metadata) {
 }
 
 function injectRemixScript(html) {
-  return html.replace(
-    /<\/head/,
-    `<script src="${REMIX_SCRIPT}" type="text/javascript"></script>\n$&`
-  );
+  // Check if <head> tag exists
+  var retInject = ``;
+  const dom = new JSDOM(html);
+
+  // If they exist, inject remix script before closing </head> tag
+  if (dom.window.document.querySelector(`head`)) {
```

Comment:
Note that JSDOM is a spec-compliant DOM parser, which means that no matter what code you throw into it, it generates a proper DOM, following the `<html><head/><body/></html>` pattern. As such, we no longer need to check for a head element, and we don't have to use strings: we can now solve this with pure DOM operations:

```
function injectRemixScript(html) {
  // We should check if this constructor can every throw. If it can,
  // we need to make sure we catch that.
  const dom = new JSDOM(html);

  // Here, as well, we should check the jsdom API to make sure that
  // there are no edge cases that can lead to a successful call to
  // the JSDOM constructor, but errors trying to access "window"
  const document = dom.window.document;

  // because we're using a spec-compliant DOM parser,
  // we are now guaranteed to have a `<head>` element.
  const head = document.head;

  // We can inject the remix script using standard DOM operations now:
  const remixScript = document.createElement("script");
  remixScript.src = REMIX_SCRIPT;
  head.appendChild(remixScript);

  // and we can then get the DOM-compliant code using the 
  // "outerHTML" property, which is supported by all modern
  // browsers: http://caniuse.com/#search=outerHTML
  const updatedHTML = document.querySelector("html").outerHTML;
  return updatedHTML;
}
```